nixos/modules/services/networking/onedrive.nix

```diff
@@ -0,0 +1,88 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.onedrive;
+
+  onedriveLauncher =  pkgs.writeShellScriptBin
+    "onedrive-launcher"
+    ''
+      # XDG_CONFIG_HOME is not recognized in the environment here.
+      if [ -f $HOME/.config/onedrive-launcher ]
+      then
+        # Hopefully using underscore boundary helps locate variables
+        for _onedrive_config_dirname_ in $(cat $HOME/.config/onedrive-launcher | grep -v '[ \t]*#' )
+        do
+          systemctl --user start onedrive@$_onedrive_config_dirname_
+        done
+      else
+        systemctl --user start onedrive@onedrive
+      fi
+    ''
+  ;
+
+  # Poor man's type/typo safety
+  systemdTargetList = [
+    "network-online"  
+    "multi-user"
+  ];
+
+  systemdTargets = builtins.listToAttrs
+  (
+    map
+      ( st:{ name = st; value = st + ".target" ; } )
+      systemdTargetList
+  );
+
+
+in {
+  ### Documentation
+  meta.doc = ./onedrive.xml;
+
+  ### Interface 
+
+  options.services.onedrive = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = "Enable OneDrive service";
+    };
+
+     package = lib.mkOption {
+       type = lib.types.package;
+       default = pkgs.onedrive;
+       defaultText = "pkgs.onedrive";
+       example = lib.literalExample "pkgs.onedrive";
+       description = ''
+         OneDrive package to use.
+       '';
+     };
+  };
+### Implementation
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.user.services."onedrive@" = {
+      description = "Onedrive sync service";
+
+      serviceConfig = {
+        Type = "simple";
```

Comment:
See if it works with `exec` instead.

Reply:
Could you explain what you mean by this? I did try exec but looking further there is no `Type=exec`. All types- https://www.freedesktop.org/software/systemd/man/systemd.service.html

Reply:
I've left the type as is. Couldn't find an `exec` type.